### ISSUE TITLE
can't catch bad substitution

### ISSUE THREAD

**Issue by kolokd** [koalaman/shellcheck#1956]
[kolokd ~/t/sh]$ cat loop.sh 
#!/bin/sh
for l in ${~/doc/*}
do
	echo "$l"
done
[kolokd ~/t/sh]$ shellcheck loop.sh 
[kolokd ~/t/sh]$ bash loop.sh 
loop.sh: line 2: ${~/doc/*}: bad substitution
[kolokd ~/t/sh]$ mksh loop.sh 
loop.sh[5]: ${~/doc/*}: bad substitution
[kolokd ~/t/sh]$ dash loop.sh 
loop.sh: 2: loop.sh: Bad substitution

**Comment by kolokd** [comment]
hi, response

**Comment by kolokd** [comment]
@koalaman please response

**Comment by koalaman** [comment]
You're right, ShellCheck currently does not flag this with a Posix warning as it should
